src/main/java/com/arcanewarrior/disguises/events/DisguiseEvents.java

```diff
@@ -15,34 +15,38 @@
 import net.minestom.server.network.packet.client.play.ClientInteractEntityPacket;
 import net.minestom.server.network.packet.client.play.ClientPlayerPositionPacket;
 import net.minestom.server.network.packet.client.play.ClientPlayerRotationPacket;
-import net.minestom.server.network.packet.server.play.EntityAnimationPacket;
-import net.minestom.server.network.packet.server.play.EntityPositionPacket;
-import net.minestom.server.network.packet.server.play.EntityRotationPacket;
+import net.minestom.server.network.packet.server.play.*;
 import net.minestom.server.utils.PacketUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class DisguiseEvents {
+public final class DisguiseEvents {
     private final DisguiseManager parentManager;
     private final Logger logger = LoggerFactory.getLogger(DisguiseEvents.class);
+
     public DisguiseEvents(DisguiseManager manager) {
         this.parentManager = manager;
     }
 
-    public void registerAll(EventNode<Event> node) {
+    public void registerAll(EventNode<PlayerEvent> node) {
         // Player Events
         EventNode<PlayerEvent> playerParent = EventNode.type("disguise-player-events", EventFilter.PLAYER);
         playerParent.addListener(PlayerPacketEvent.class, this::handlePlayerPacketInput);
         playerParent.addListener(PlayerPacketOutEvent.class, this::playerEntityAnimation);
+        playerParent.addListener(PlayerPacketOutEvent.class, this::playerTeleport);
```

Comment:
I would prefer that we have 1 listener for PlayerPacketOutEvent.class, instead of the 10-20 we would likely need to cover all relevant packet translation, and combine the various methods into 1 large one. If you wish you continue with the separated methods, please instead have a method that delegates the details of the event to the appropriate handler method.

Reply:
I will do it later too